core/src/main/scala/io/projectglow/transformers/normalizevariants/VariantNormalizer.scala

```diff
@@ -16,318 +16,182 @@
-          nTrimmedBasesBeforeNextPadding -= 1
-        }
 
-        alleles = alleles.map { a =>
-          Allele.create(
-            padSeq ++ a
-              .getBaseString()
-              .dropRight(nTrimmedBasesBeforeNextPadding + 1)
-              .getBytes(),
-            a.isReference
-          )
-        }
+          nTrimmedBasesBeforeNextPadding += 1
 
-        trimSize += nTrimmedBasesBeforeNextPadding + 1
+        }
 
-        newStart -= nPadBases
+        firstBaseFromRightInRefAllele = allAlleles(0)(
+          allAlleles(0).length - nTrimmedBasesBeforeNextPadding - 1
+        )
+      }
 
-        nTrimmedBasesBeforeNextPadding = 0
+      // trim from left
+      var nLeftTrimBases = 0
+      var firstBaseFromLeftInRefAllele = allAlleles(0)(nLeftTrimBases)
+      val minAlleleLength = allAlleles.map(_.length).min
 
-      } else {
+      while (nLeftTrimBases < minAlleleLength - nTrimmedBasesBeforeNextPadding - 1
+        && allAlleles.forall(_(nLeftTrimBases) == firstBaseFromLeftInRefAllele)) {
 
-        nTrimmedBasesBeforeNextPadding += 1
+        nLeftTrimBases += 1
 
+        firstBaseFromLeftInRefAllele = allAlleles(0)(nLeftTrimBases)
       }
 
-      firstAlleleBaseFromRight = alleles(0).getBases()(
-        alleles(0).length
-        - nTrimmedBasesBeforeNextPadding - 1
-      )
-    }
-
-    // trim from left
-    var nLeftTrimBases = 0
-    var firstAlleleBaseFromLeft = alleles(0).getBases()(nLeftTrimBases)
-    val minAlleleLength = alleles.map(_.length).min
-
-    while (nLeftTrimBases < minAlleleLength - nTrimmedBasesBeforeNextPadding - 1
-      && alleles.forall(_.getBases()(nLeftTrimBases) == firstAlleleBaseFromLeft)) {
+      allAlleles = allAlleles.map { a =>
+        a.drop(nLeftTrimBases)
+          .dropRight(nTrimmedBasesBeforeNextPadding)
+      }
 
-      nLeftTrimBases += 1
+      trimSize += nTrimmedBasesBeforeNextPadding
 
-      firstAlleleBaseFromLeft = alleles(0).getBases()(nLeftTrimBases)
-    }
-
-    alleles = alleles.map { a =>
-      Allele.create(
-        a.getBaseString()
-          .drop(nLeftTrimBases)
-          .dropRight(nTrimmedBasesBeforeNextPadding)
-          .getBytes(),
-        a.isReference
-      )
+      newStart += nLeftTrimBases
 
+      flag = if (trimSize == 0 && nLeftTrimBases == 0) {
+        FLAG_UNCHANGED
+      } else {
+        FLAG_CHANGED
+      }
     }
 
-    trimSize += nTrimmedBasesBeforeNextPadding
-
-    AlleleBlock(
-      alleles,
-      newStart + nLeftTrimBases,
-      unalignedAlleleBlock.end - trimSize
+    InternalRow(
+      newStart,
+      end - trimSize,
+      UTF8String.fromString(allAlleles(0)),
+      ArrayData.toArrayData(allAlleles.tail.map(UTF8String.fromString(_))),
+      UTF8String.fromString(flag)
     )
 
   }
 
   private val PAD_WINDOW_SIZE = 100
 
+  val FLAG_ERROR = "Error"
+  val FLAG_CHANGED = "Changed"
+  val FLAG_UNCHANGED = "Unchanged"
+
+  val normalizedStartField = StructField("normalizedStart", LongType)
```

Comment:
Hm, I don't love adding the `normalized` prefix to all of these column names. Since this expression returns a struct, it's simple to keep the fields grouped together. We can use the normal names (`start`, `end`, etc). In the case where we don't replace columns, we can just call this struct `normalizationResult` or something like that.

Reply:
Changed to what you mention.